### ISSUE TITLE
 Inconsistent DateTime format returned from the HTTP API with varying precision in the same Measurement

### ISSUE THREAD

**Issue by michaelzg** [influxdata/influxdb#7478]
## Inconsistent DateTime format returned from the HTTP API with varying precision in the same Measurement

__System info:__ 
* Version: 1.0.0
* Commit: 37992377a55fbc138b2c01edd4deffed64b53989
* amd64, linux, go1.6.2

__Steps to reproduce:__

*  1. Make a database, call it `dbTest` for the sake of this example
*  2. Insert millisecond precision data with CLI to the `test` measurement
```
INSERT into autogen test,prec=millis value=1 1465934559444000000
```
* 3. Insert second precision data with CLI to the same `test` measurement
```
INSERT into autogen test,prec=second value=2 1465934561000000000
```
* 4. Query for these using the HTTP API:

The second precision data returns timestamp `2016-06-14T20:02:41Z` , notice *no milliseconds*.
```
localhost:8086/query?pretty=true&db=dbTest&q=select * from test where prec='second'
```

The millisecond precision data returns timestamp `2016-06-14T20:02:39.444Z`, notice milliseconds
```
localhost:8086/query?pretty=true&db=dbTest&q=select * from test where prec='millis'
```

Another view of this is issue is querying for both the measurements and seeing two different DateTime formats returned in the same response. 
```
localhost:8086/query?pretty=false&db=dbTest&q=select * from test
```

__Expected behavior:__ 
The timestamp format returned from the HTTP API should be standardized. One scheme would be to return the format corresponding to the *most precise* timestamp precision within the measurement. So, in the example above, the second precision data should have returned with padded zeros to the millisecond precision: `2016-06-14T20:02:41.000Z` because the `test` measurement contains millisecond precision timestamps. 

This is important for DateTime parsers consuming the HTTP API that expects a consistent format.

**Comment by jwilder** [comment]
We do use a standard format.  The timestamps are  [RFC3339](https://tools.ietf.org/html/rfc3339) formatted.  [Section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6) indicates that the fractional time part is optional:

```
 partial-time    = time-hour ":" time-minute ":" time-second
                     [time-secfrac]
```

Can you configure your DataTime parser to parse RFC3339 timestamps?

**Comment by michaelzg** [comment]
Thanks for the quick reply and reference.

Using joda, I found I had to use [ISODateTimeFormat.datetimeNoMillis](http://joda-time.sourceforge.net/apidocs/org/joda/time/format/ISODateTimeFormat.html#dateTimeNoMillis()) versus [ISODateTimeFormat.datetime](http://joda-time.sourceforge.net/apidocs/org/joda/time/format/ISODateTimeFormat.html#dateTime()) 

It was recently pointed out to me that the HTTP API takes in an `epoch` field. Using that can workaround this issue by specifying read precision for timestamps.

I feel like this is resolved, thank you :)
